### ISSUE TITLE
cmake build creates only lib64 not lib 

### ISSUE THREAD

**Issue by PrometheusPi** [openPMD/openPMD-api#792]
**Description of the documentation bug**
If installing openPMD-api as described in the docs, a `lib64` directory is created not a `lib` directory as stated in the [documentation line 180](https://github.com/openPMD/openPMD-api/blob/dev/docs/source/install/install.rst#L180).  

**How to reproduce it**
When installing openPMD-api via:
```bash
mkdir -p ~/src ~/lib
cd ~/src
git clone https://github.com/openPMD/openPMD-api.git
cd openPMD-api
mkdir build && cd build
cmake .. -DopenPMD_USE_MPI=ON -DCMAKE_INSTALL_PREFIX=~/lib/openPMD-api
make -j $(nproc) install
```
The path `$HOME/lib/openPMD-api/lib` is not created but only `$HOME/lib/openPMD-api/lib64` - thus can be reproduced on summit/ORNL and hemera/HZDR. 

**Expected behavior**
I would have expected the `lib` path to be created or to be a symbolic link to `lib64`. 

**Software Environment**
 - version of openPMD-api: current dev
 - installed openPMD-api via:  from source
 - operating system: 
hemera:
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

summit:
```
NAME="Red Hat Enterprise Linux Server"
VERSION="7.6 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.6"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.6 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.6:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.6
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.6"
```
 - machine: yes - clusters you have also access too 
 - name and version of Python implementation: hemera: python/3.6.5 - summit: none
 - version of HDF5: hemera: hdf5-parallel/1.8.20-cuda102 - summit: hdf5/1.10.3  
 - version of ADIOS1: hemera: adios/1.13.1-cuda102 - summit: adios/1.13.1-py2
 - version of ADIOS2: none
 - name and version of MPI: hemera: openmpi/2.1.2-cuda102  - summit: spectrum-mpi/10.3.1.2-20200121

**Possible solution**
Can be solved by exporting 
```diff
- export LD_LIBRARY_PATH=$HOME/somepath/lib:$LD_LIBRARY_PATH
+ export LD_LIBRARY_PATH=$HOME/somepath/lib64:$LD_LIBRARY_PATH
```
instead.

**Comment by ax3l** [comment]
Hi Richard, thank you for opening and documenting this.

A bit inconsistently in the Unix/Linux world, either lib or lib64 can be the right library path, depending if the distribution supports 32bit and 64bit libraries alongside each other:
- https://cmake.org/cmake/help/v3.18/module/GNUInstallDirs.html
- https://unix.stackexchange.com/questions/74646/difference-between-lib-lib32-lib64-libx32-and-libexec

There is no real reason to change that default of the operating system - which is picked up by CMake here, it does the right thing by default. If you really want to overwrite it you can pass `-DCMAKE_INSTALL_LIBDIR=lib` nonetheless, which does no harm for non-system prefixes.

In the documentation, the last section is just an example for very manual usage and might need to be clarified, thanks for raising this. Indeed, if you use CMake with RPATHS enabled for a downstream project you never have to set `LD_LIBRARY_PATH`. Just point to the install prefix with `CMAKE_PREFIX_PATH` for the build and you are set.

I think adding a short note on lib64 could clarify this, what do you think?

**Comment by PrometheusPi** [comment]
@ax3l I agree, a brief comment that of x86x64 systems, `lib64` should be used instead of `lib` would be enough in the documentation. I will provide a fix.

**Comment by ax3l** [comment]
Thank you again. Fixed via #793
